enhancements/opm-content-management.md

```diff
@@ -0,0 +1,145 @@
+---
+title: opm-content-management 
+authors:
+  - "@anik120"
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: yyyy-mm-dd
+last-updated: yyyy-mm-dd
+status: implementable
+---
+
+# OPM Content Management
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+
+## Summary
+
+This enhancement is a proposal for extending the existing capabilites of the `opm` command line utility, to provide more fine grained control over the management of catalog indexes.   
+
+`opm` will have significant improvement to UX for editing an index (removing individual bundles from a package, removing channels from a package). Indexes can also be queried directly for inspection, and update graph visualization for individual packages will be added. Bundle validation without a deamon will is also proposed in this enhancement. 
+
+## Motivation
+
+As more and more operators are migrated to the new bundle format, and more index images are created for Operator Catalogs, creators of catalogs of operators, such as the Red Hat pipeline responsible for maintaining the `redhat-operators` catalog, should have more fine grained control over the management of the catalogs using the `opm` tooling.  
+
+### Goals
+
+- Allow `opm` users to perform bundle validations in environments without a Docker daemon. Bundle validation also includes validation of bundle image labels against `metadata/annotations.yaml`
+- Allow `opm` users to safely edit indexes, which includes removing bundle version from a package in an index, and removing a channel from a bundle. 
+- Allow `opm` users to inspect catalog content and return the update graph of Operators in machine-readable format.
+
```

Comment:
IIRC we also want to define the source of truth for an index so that it can be rebuilt. IMO the transactions that are applied to the DB that don't come from bundle labels / annotations should be logged or output somewhere.

Reply:
+1 understanding provenance and reproducibility of distributed artifacts is key.

Reply:
@gallettilance @mhrivnak the state of the index is already recorded in a database, which is also edited when content in the index is edited. The information on the database is sufficient to rebuild the index with, unless there's other reasoning/s to support logging of the transactions.

Reply:
@anik120 I suggest to plug here your parallel work on declarative index configs, to form the base of reproducible catalog/index creation. The goal being:

- Allow `opm` users to create a catalog / index reproducible from scratch using an on-disk index representation in human-friendly and machine-readable text format